src/core/useInfiniteScroll.ts

```diff
@@ -0,0 +1,23 @@
+import { useCallback, useRef } from "react";
+
+export const useInfiniteScroll = (loadMore: () => void, dependencies = []) => {
```

Comment:
Do you know if how better it is to use`IntersectionObserver` over using https://github.com/danbovey/react-infinite-scroller?

I used this library which is pretty working well.

Reply:
It's small enough `2.2KB gzipped`.

https://github.com/danbovey/react-infinite-scroller/blob/master/src/InfiniteScroll.js#L38-L40

Reply:
I tried it, but it was messing with my layout. And when all the results fit the page it failed to trigger a load of the next "page".